src/he/pmtud.h

```diff
@@ -0,0 +1,140 @@
+/**
+ * Lightway Core
+ * Copyright (C) 2023 Express VPN International Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/**
+ * @file pmtud.h
+ * @brief Path MTU Discovery functions
+ *
+ */
+
+#ifndef PMTUD_H
+#define PMTUD_H
+
+#include "he.h"
+
+/// The maximum value of the PROBE_COUNT counter. MAX_PROBES represents the limit for the number of
+/// consecutive probe attempts of any size.
+#define MAX_PROBES 3
+
+/// The smallest PMTU the discovery process will attempt to use
+#define MIN_PLPMTU 512
+
+/// The largest PMTU the discovery process will attempt to use
+#define MAX_PLPMTU                                                                      \
+  (HE_MAX_WIRE_MTU - HE_IPV4_HEADER_SIZE - HE_UDP_HEADER_SIZE - sizeof(he_wire_hdr_t) - \
+   HE_WOLF_MAX_HEADER_SIZE - sizeof(he_msg_data_t))
```

Comment:
This seems like it is somehow subtly related to `HE_PACKET_OVERHEAD`? Should the two be somehow consistent?

TBH I mostly mention it since I don't really grok how `HE_PACKET_OVERHEAD` was arrived at and maybe having done the PMTU stuff you are currently in a position to understand it....

Reply:
HE_PACKET_OVERHEAD uses a deprecated message header so I don't think it's proper to reuse it. 

The use cases of HE_PACKET_OVERHEAD mostly for preventing sending too large data to wolfssl DTLS layer. We can probably deprecate those use cases with PMTUD and Fragmentation features.

Reply:
I was thinking more "consolidate" than reuse, but I agree the need should go away with PMTUD.